README.md

```diff
@@ -46,23 +79,66 @@ const resolved = await resolver.resolve({
   }
 });
 
-console.log(resolved.result);
-
-// ==> outputs the original object, with local refs resolved and replaced
-//
-// {
-//   user: {
-//     name: 'json'
-//   },
-//   models: {
-//     user: {
-//       name: 'john'
-//     }
-//   }
-// }
+// ==> result is the original object, with local refs resolved and replaced
```

Comment:
I get what you mean here, however I'd say "it's a new object with local refs resolved and replaced"

Reply:
Not a new object though, it's better than that :). Added a few new test assertions to demonstrate the immutability principles, check them out: https://github.com/stoplightio/json-ref-resolver/blob/feat/ref-map/src/__tests__/resolver.spec.ts#L139-L147

Structural sharing is particularly important for frontend applications like react, that can take advantage of it to re-render less parts of the screen since `===` will remain true for much of the object (anything that was not a ref).